crates/oxc_linter/src/rules/eslint/yoda.rs

```diff
@@ -0,0 +1,1160 @@
+use oxc_ast::{
+    ast::{
+        BinaryExpression, BinaryOperator, Expression, LogicalExpression, LogicalOperator,
+        UnaryOperator,
+    },
+    AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_ecmascript::ToBigInt;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+use regex::Regex;
+
+use crate::{context::LintContext, rule::Rule, utils::is_same_reference, AstNode};
+
+fn yoda_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Require or disallow \"Yoda\" conditions")
+        .with_help("Expected literal to be on the opposite side of the operator.")
+        .with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct Yoda {
+    never: bool,
+    except_range: bool,
+    only_equality: bool,
+}
+
+declare_oxc_lint!(
+    /// ## What it does
+    ///
+    /// Require or disallow "Yoda" conditions.
+    /// This rule aims to enforce consistent style of conditions which compare a variable to a literal value. The literal value must always come first in the comparison.
```

Comment:
is "The literal value must always come first in the comparison." correct? shouldn't it be the other way round?

Reply:
deleted. https://github.com/oxc-project/oxc/pull/7559/commits/abecc1a71c09dd523cd306df8617b33b5d963b8f
(I'm not sure why this sentence was inserted.. maybe copilot did..?)